File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/base/ReportBase.java
package org.nirvanasms.api.core.base;

import java.util.List;

import org.nirvanasms.api.core.entity.NumberReport;
import org.nirvanasms.api.core.interfaces.IReport;
import org.nirvanasms.api.core.utility.ProcessResult;

public abstract class ReportBase implements IReport {
	@Override
	public ProcessResult<List<NumberReport>> getReport(UserBase apiUser, String messageId){
		return null;
	}
}

File: NirvanaSMS.Api.CSharp/Core/Base/ReportBase.cs
using NirvanaSMS.Api.Core.Interface;
using System.Collections.Generic;
using NirvanaSMS.Api.Core.Entity;
using NirvanaSMS.Api.Core.Utility;

namespace NirvanaSMS.Api.Core.Base
{
    public abstract class ReportBase : IReport
    {
        public virtual ProcessResult<List<NumberReport>> GetReport(UserBase apiUser, string messageId)
        {
            return null;
        }
    }
}
